content/en/docs/Writing policies/variables.md

```diff
@@ -182,6 +182,20 @@ Because Helm executes its templating routine prior to Kyverno, a Kyverno policy
 {{`{{ request.userInfo.username }}`}}
 ```
 
+For Kyverno variables which use more complex JMESPath expressions including [existence checks](/docs/writing-policies/jmespath/#non-existence-checks), elements such as empty objects or arrays may also need to be escaped even within the overall expression. For example, the value of the below `value` field
+
+```
+value: "{{ element.securityContext.capabilities.drop[].to_upper(@) || `[]` }}"
+```
+
+would need to become
+
+```
+value: {{ `"{{ element.securityContext.capabilities.drop[].to_upper(@) || `}}`[]`{{` }}"` }}
```

Comment:
This is new to me, how to read this expression?

Reply:
It's the same as above just escaped. Multiple sets of the braces have to be used to avoid Helm trying to interpret `[]`.